siac/parse_test.go

```diff
@@ -33,9 +35,40 @@ func TestParseSize(t *testing.T) {
 		{"1.2345KB", "1234", nil},
 	}
 	for _, test := range tests {
-		res, err := parseSize(test.in)
+		res, err := parseFilesize(test.in)
 		if res != test.out || err != test.err {
-			t.Error(fmt.Sprintf("parseSize(%v): expected %v %v, got %v %v", test.in, test.out, test.err, res, err))
+			t.Errorf("parseFilesize(%v): expected %v %v, got %v %v", test.in, test.out, test.err, res, err)
+		}
+	}
+}
+
+func TestCurrencyUnits(t *testing.T) {
+	tests := []struct {
+		in, out string
+	}{
+		{"1", "1 H"},
+		{"1000", "1000 H"},
+		{"100000000000", "100000000000 H"},
+		{"1000000000000", "1 pS"},
+		{"1234560000000", "1.2346 pS"},
+		{"12345600000000", "12.346 pS"},
+		{"123456000000000", "123.46 pS"},
```

Comment:
4 is probably enough. 3 doesn't seem sufficient because if you have a 3-digit number you have no decimal at all.